netbox/extras/templatetags/plugins.py

```diff
@@ -0,0 +1,71 @@
+from django import template as template_
+from django.conf import settings
+from django.utils.safestring import mark_safe
+
+from extras.registry import registry
+
+register = template_.Library()
+
+
+def _get_registered_content(obj, method, template_context):
+    """
+    Given an object and a PluginTemplateExtension method name and the template context, return all the
+    registered content for the object's model.
+    """
+    html = ''
+    context = {
+        'obj': obj,
+        'request': template_context['request'],
+        'settings': template_context['settings'],
+        'config': {},  # Defined per-plugin
+    }
+
+    model_name = obj._meta.label_lower
+    template_extensions = registry['plugin_template_extensions'].get(model_name, [])
+    for template_extension in template_extensions:
+
+        # Update context with plugin-specific configuration parameters
+        plugin_name = template_extension.__module__.split('.')[0]
+        context['config'] = settings.PLUGINS_CONFIG.get(plugin_name)
+
+        instance = template_extension(context)
+        try:
+            content = getattr(instance, method)()
+        except NotImplementedError:
+            # This content renderer class does not define content for this method
+            continue
```

Comment:
Can we optimize this? Currently we're populating context data, instantiating a class, and calling a method just to determine if the method is actually defined. This approach does allow for a tidy plugin development experience, but it also imposes a lot of unnecessary overhead when rendering templates.

I'm not sure what a better approach would look like. I think ideally we'd call `hasattr()` on the class first to determine whether any further action is needed.

Reply:
Personally I think it is a premature optimization. The time complexity of this is the same either way: O(N) over the list of registered classes. We could create a larger dict in registry per class method, but is it really worth it?

Reply:
Under the current approach, each applicable PluginTemplateExtension is instantiated four times per view (one for each available method). It's almost certain we'll add more methods at some point in the future, so we might as well get ahead of it now.

Reply:
Came up with a simple shortcut: if the sublcass' method is the same as the parent class', skip it.

```python
if getattr(template_extension, method) == getattr(PluginTemplateExtension, method):
    continue
```